File: models/OilChange.py
from models.Serializer import Serializer


class OilChange(Serializer):
    fields = [
        "id",
        "vehicle",
        "odometer",
        "date",
    ]

    def __init__(self, id=None, vehicle=None, odometer=None, date=None):
        self.id = id
        self.vehicle = vehicle
        self.odometer = odometer
        self.date = date

File: src/cp.models/oilChange.js
const {Model, DataTypes} = require('sequelize')
const sequelize = require('./connection')

class OilChange extends Model {}

OilChange.init(
  {
    id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      autoIncrement: true,
      primaryKey: true,
    },
    vehicle: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: 'vehicle',
      onDelete: 'CASCADE',
    },
    odometer: {
      type: DataTypes.INTEGER,
    },
    date: {
      type: DataTypes.DATE,
      allowNull: false,
      defaultValue: DataTypes.NOW,
    },
  },
  {sequelize, modelName: 'oil_change', timestamps: false},
)

module.exports = OilChange
